### PR TITLE
New branch

### PR DESCRIPTION
**Pull request by danilashtefan**
SonarCloud issues corrected in Torpedo.java

### CODE REVIEW

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 1]
```diff
@@ -13,10 +13,12 @@
   private double FAILURE_RATE = 0.0; //NOSONAR
 
   private int torpedoCount = 0;
+  Random generator = new Random();
```
Comment (DinuBotan) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:16]: Excellent work here!

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 2]
```diff
@@ -13,10 +13,12 @@
...
 
   public TorpedoStore(int numberOfTorpedos){
     this.torpedoCount = numberOfTorpedos;
 
+
```
Comment (DinuBotan) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java]: This was unnecessary..

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 3]
```diff
@@ -30,18 +32,18 @@ public TorpedoStore(int numberOfTorpedos){
 
   public boolean fire(int numberOfTorpedos){
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-      new IllegalArgumentException("numberOfTorpedos");
+      throw new IllegalArgumentException("numberOfTorpedos");
```
Comment (DinuBotan) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:34]: Keep up the good work!

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 4]
```diff
@@ -30,18 +32,18 @@ public TorpedoStore(int numberOfTorpedos){
...
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
+
     double r = generator.nextDouble();
 
     if (r >= FAILURE_RATE) {
       // successful firing
-      this.torpedoCount =- numberOfTorpedos;
+      this.torpedoCount -= numberOfTorpedos;
```
Comment (DinuBotan) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:45]: Well done!

**Review by DinuBotan — CHANGES_REQUESTED** [review]
Good job overall, but you have an unnecessary step as well.

**Review by DinuBotan — APPROVED** [review]

**Comment by micskeiz** [comment]
Nice work.

**Pull request merged**